qiskit/transpiler/passes/basis/basis_translator.py

```diff
@@ -130,6 +130,9 @@ def run(self, dag):
         qarg_local_basis_transforms = {}
         for qarg, extra_source_basis in incomplete_source_basis.items():
             expanded_target = target_basis | qarg_with_incomplete[qarg]
+            if len(qarg) > 1:
+                for qubit in qarg:
+                    expanded_target |= qarg_with_incomplete.get((qubit,), set())
```

Comment:
Is there a risk here of pulling in instructions that are not supported on all qubits or set of qubits, or does this only fetch 1q-gates ?

Reply:
yeah there a potential issue around this, but it's the best solution available. See @georgios-ts's comment here: https://github.com/Qiskit/qiskit-terra/pull/7227#issuecomment-984760416 I had an improvement on this older revision here: https://github.com/Qiskit/qiskit-terra/commit/7e8adf47754279f05a9aefe4ad8b0a1232586143 and added a known issue release note for it: https://github.com/Qiskit/qiskit-terra/pull/7227/commits/ed0025a4dd19394d95748e2b8d9f535e2fd066b8